src/benchmarks/micro/corefx/System.Text.Json/Utf8JsonReader/Utf8JsonReader.Comments.Tests.cs

```diff
@@ -0,0 +1,138 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace System.Text.Json.Tests
+{
+    [BenchmarkCategory(Categories.CoreFX, Categories.JSON)]
+    public partial class Utf8JsonReaderCommentsTests
+    {
+        [Params(JsonCommentHandling.Skip, JsonCommentHandling.Allow)]
+        public JsonCommentHandling CommentHandling;
+
+        // 0 => single segment
+        [Params(0, 100)]
+        public int SegmentSize;
+
+        private bool MultiSegment => SegmentSize != 0;
+        private byte[] _jsonPayload;
+        private ReadOnlySequence<byte> _jsonPayloadSequence;
+
+        [ParamsSource(nameof(TestCaseValues))]
```

Comment:
just a hint for future: BenchmarkDotNet has a `[ParamsAllValues]` attribute that expands given `[Params]` property with all enum values